### PR TITLE
Add C++ example with OpenCV and Eigen

### PR DESCRIPTION
**Pull request by roym899**
TODOs
- [x] log `std::vector<Eigen::Vector3f>`
- [x] log `Eigen::MatrixX3f`
- [x] log poses and camera
- [x] log image from OpenCV (wait for Image archetype constructors) 
- [x] clean up code
- [x] remove hardcoded paths in `CMakeLists.txt`  (wait for better way to install)
- [x] add install instructions for Rerun and other dependencies required to build this example

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+#include <opencv2/core.hpp>
+#include <opencv2/highgui.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <rerun.hpp>
+
+template <>
+struct rerun::ComponentBatchAdapter<rerun::components::Position3D, std::vector<Eigen::Vector3f>> {
+    // Sanity check that this is binary compatible.
+    static_assert(sizeof(components::Position3D) == sizeof(Eigen::Vector3f));
+    static_assert(alignof(components::Position3D) <= alignof(Eigen::Vector3f));
```
Comment (Wumpf) [inline src/main.cpp:13]: ComponentBatch should have `borrow_compatible` which does these and a few other checks checks for you on 💡

**Comment by emilk** [comment]
I'm gonna see if we can use `pixi` to install eigen, opencv, and arrow

**Comment by emilk** [comment]
`pixi run run` just works™️ now. I love pixi!

**Comment by emilk** [comment]
Code will become a lot simpler once this is merged:
* https://github.com/rerun-io/rerun/pull/3846

**Review by Wumpf — APPROVED** [review]
some more fine tuning to do, in particular with the adapters, but good start

**Pull request merged**